### DISCUSSION TITLE
How do you get clean database state after every test?

### DISCUSSION THREAD

**Discussion by jerrytomato** [TomGrozev/arangox_ecto#69] [Q&A] [answered]
I have a problem with my tests. When I try to create a users fixture, I'm getting a unique constraint violation because the database state is not cleaned up after every test. Is there an easy way to get around this? I suppose the best way would be to create a Sandbox adapter, but is there a quicker fix so I can get on to writing some tests?

I'd love some direction, and of course I'm more than happy to write a testing section in the docs as an update. Thank you for this library!

**Comment by TomGrozev** [comment]
I managed to get it setup working quite well in the tests for this repo. Checkout [https://github.com/TomGrozev/arangox_ecto/tree/develop/test](https://github.com/TomGrozev/arangox_ecto/tree/develop/test)

Some of the reseting database functions are a little lower level due to the need to be actually testing the repo but you can abstract them into using the ArangoXEcto functions. That's what I have done in my projects using this package.

Because it works just like a normal Ecto adapter, testing should be generally the same as the other Ecto adapters. If you reckon some more documentation on this is needed, by all means a PR is appreciated.

**Reply by jerrytomato** [reply]
Thank you! Is it the setup_all and setup call that you're referring to here?
  https://github.com/TomGrozev/arangox_ecto/blob/b959a7e856f3719678308a1b5af55469537e43fe/test/arangox_ecto_test.exs#L22C3-L86C6

Do you see something like a port of Ecto.Adapters.SQL.Sandbox being useful for this project?

**Reply by TomGrozev** [reply]
Pretty much just those two. I have historically just preferred to use the actual db rather than a sandbox. I just don't have a need for it right now but do understand its utility. However if you reckon it would be useful for this project I would encourage a PR.

**Reply by jerrytomato** [reply]
Yeah I understand your point of view. However I'm a big fan of this project and I think that some sort of drop-in replacement for the sandbox would be good for adoption. I'd love to see more people using elixir and arango together. I'll think it through and if it looks like something I could do relatively easily, I'll reach out.

**Reply by TomGrozev** [reply, marked as answer]
This is now supported in version 2.0 where a sandbox exists for tests.
